libcontainer/console.go

```diff
@@ -17,57 +15,3 @@ type Console interface {
 	// Fd returns the fd for the master of the pty.
 	File() *os.File
 }
-
-const (
-	TerminalInfoVersion uint32 = 201610041
-	TerminalInfoType    uint8  = 'T'
-)
-
-// TerminalInfo is the structure which is passed as the non-ancillary data
-// in the sendmsg(2) call when runc is run with --console-socket. It
-// contains some information about the container which the console master fd
-// relates to (to allow for consumers to use a single unix socket to handle
-// multiple containers). This structure will probably move to runtime-spec
-// at some point. But for now it lies in libcontainer.
-type TerminalInfo struct {
```

Comment:
To be honest I didn't benchmark it, but it seemed like a useful optimisation because it allowed us to also pass around information about the container to the caller (rather than just a file descriptor). But if you feel that's not necessary, we can implement it in `cri-o` with a new socket for every `conmon` and presumably you can do the same thing in `containerd-shim`.